lake/api/remote.go

```diff
@@ -0,0 +1,244 @@
+package api
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"net/http"
+	"strings"
+
+	"github.com/brimdata/zed/api"
+	"github.com/brimdata/zed/api/client"
+	"github.com/brimdata/zed/api/queryio"
+	"github.com/brimdata/zed/driver"
+	"github.com/brimdata/zed/expr/extent"
+	"github.com/brimdata/zed/lake"
+	"github.com/brimdata/zed/lake/index"
+	"github.com/brimdata/zed/lake/journal"
+	"github.com/brimdata/zed/order"
+	"github.com/brimdata/zed/zbuf"
+	"github.com/brimdata/zed/zio"
+	"github.com/brimdata/zed/zio/anyio"
+	"github.com/brimdata/zed/zio/zngio"
+	"github.com/brimdata/zed/zio/zsonio"
+	"github.com/brimdata/zed/zson"
+	"github.com/segmentio/ksuid"
+)
+
+var _ Interface = (*RemoteSession)(nil)
+
+type RemoteSession struct {
+	conn *client.Connection
+}
+
+func OpenRemoteLake(ctx context.Context, host string) (*RemoteSession, error) {
+	return &RemoteSession{
+		conn: newConnection(host),
+	}, nil
+}
+
+func newConnection(host string) *client.Connection {
+	if !strings.HasPrefix(host, "http") {
+		host = "http://" + host
+	}
+	return client.NewConnectionTo(host)
+}
+
+func (r *RemoteSession) ScanPools(ctx context.Context, zw zio.Writer) error {
+	res, err := r.conn.ScanPools(ctx)
+	if err != nil {
+		return err
+	}
+	defer res.Body.Close()
+	zr := zngio.NewReader(res.Body, zson.NewContext())
+	return zio.CopyWithContext(ctx, zw, zr)
+}
+
+func (r *RemoteSession) LookupPool(ctx context.Context, name string) (*lake.PoolConfig, error) {
+	res, err := r.conn.ScanPools(ctx)
+	if err != nil {
+		return nil, nil
+	}
+	defer res.Body.Close()
+	format, err := api.MediaTypeToFormat(res.ContentType)
+	if err != nil {
+		return nil, err
+	}
+	zr, err := anyio.NewReaderWithOpts(res.Body, zson.NewContext(), anyio.ReaderOpts{Format: format})
+	if err != nil {
+		return nil, nil
+	}
+	for {
+		rec, err := zr.Read()
+		if rec == nil || err != nil {
+			return nil, err
+		}
+		var pool lake.PoolConfig
+		if err := zson.UnmarshalZNGRecord(rec, &pool); err != nil {
+			return nil, err
+		}
+		if pool.Name == name {
+			return &pool, nil
+		}
+	}
+}
+
+func (r *RemoteSession) CreatePool(ctx context.Context, name string, layout order.Layout, thresh int64) (*lake.PoolConfig, error) {
+	res, err := r.conn.PoolPost(ctx, api.PoolPostRequest{
+		Name:   name,
+		Layout: layout,
+		Thresh: thresh,
+	})
+	if err != nil {
+		return nil, err
+	}
+	var config lake.PoolConfig
+	unmarshal(res, &config)
+	return &config, nil
+}
+
+func (r *RemoteSession) RemovePool(ctx context.Context, pool ksuid.KSUID) error {
+	return r.conn.PoolRemove(ctx, pool)
+}
+
+func (r *RemoteSession) RenamePool(ctx context.Context, pool ksuid.KSUID, name string) error {
+	if name == "" {
+		return errors.New("no pool name provided")
+	}
```

Comment:
It doesn't seem to me that a command-line tool should rely upon the server to detect the zero value on a non-optional argument when it could easily check this here before interacting with the server.